src/utils.ts

```diff
@@ -103,7 +103,12 @@ export function createWebSocket () {
   return {
     serve,
     broadcast,
-    close: () => new Promise(resolve => wss.close(resolve))
+    close: () => {
+      // disconnect all clients
+      wss.clients.forEach(client => client.close())
```

Comment:
Just double-checking, doesn't `client.close` return a promise/callback? In this case, we could await on promises.